stripe/src/main/java/com/stripe/android/model/Stripe3ds2Fingerprint.kt

```diff
@@ -85,14 +86,36 @@ internal class Stripe3ds2Fingerprint private constructor(
         internal fun create(sdkData: StripeIntent.SdkData): Stripe3ds2Fingerprint {
             require(sdkData.is3ds2) { "Expected SdkData with type='stripe_3ds2_fingerprint'." }
 
-            return Stripe3ds2Fingerprint(
-                (sdkData.data[FIELD_THREE_D_SECURE_2_SOURCE] as String),
-                DirectoryServer.lookup((sdkData.data[FIELD_DIRECTORY_SERVER_NAME] as String)),
-                (sdkData.data[FIELD_SERVER_TRANSACTION_ID] as String),
-                DirectoryServerEncryption.create(
-                    (sdkData.data[FIELD_DIRECTORY_SERVER_ENCRYPTION] as Map<String, *>)
-                )
-            )
+            when (sdkData.data) {
+                is Either.Left -> {
+                    val (data: Map<String, *>) = sdkData.data
+                    return Stripe3ds2Fingerprint(
+                        (data[FIELD_THREE_D_SECURE_2_SOURCE] as String),
+                        DirectoryServer.lookup((data[FIELD_DIRECTORY_SERVER_NAME] as String)),
+                        (data[FIELD_SERVER_TRANSACTION_ID] as String),
+                        DirectoryServerEncryption.create(
+                            (data[FIELD_DIRECTORY_SERVER_ENCRYPTION] as Map<String, *>)
+                        )
+                    )
+                }
+                is Either.Right -> {
+                    val (data: PaymentIntent.NextActionData.SdkData) = sdkData.data
+                    require(data is PaymentIntent.NextActionData.SdkData.Use3DS2) {
```

Comment:
can we avoid `require`?

Reply:
not easily. This (old) SdkData object is pretty weakly typed so it makes things super awkward. That's part of the point of this refactor. However, this is removed in #2496 once the refactor is done.